src/ShadowDOM/wrappers/Range.js

```diff
@@ -20,23 +20,69 @@
 
   var OriginalRange = window.Range;
 
+  var ShadowRoot = scope.wrappers.ShadowRoot;
+
+  function isElement(node){
+    return node && node.nodeType == Node.ELEMENT_NODE
+  }
+
+  function getHost(node){
+    while(node && node.parentNode){
```

Comment:
fyi, I think this method can just be:
```js
var getTreeScope = scope.getTreeScope;

function getHost(node) {
  var root = getTreeScope(node).root;
  if (root instanceof ShadowRoot) return root.host;
  return null;
}
```

TreeScope caches a quick way to jump to the root of the current tree (either Document, DocumentFragment, or ShadowRoot)